llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -5614,17 +5614,98 @@ static SDValue TryMULWIDECombine(SDNode *N,
   return DCI.DAG.getNode(Opc, DL, MulType, TruncLHS, TruncRHS);
 }
 
+static SDValue matchMADConstOnePattern(SDValue X, SDValue Add) {
+  if (Add->getOpcode() != ISD::ADD)
+    return SDValue();
+
+  SDValue Y = Add->getOperand(0);
+  ConstantSDNode *Const = dyn_cast<ConstantSDNode>(Add->getOperand(1));
+  if (!Const || Const->getZExtValue() != 1)
```

Comment:
Nit. Phrasing the condition in positive terms would be more readable, IMO. 
`if (Const && Const->getZExtValue() == 1) return Y;`

Reply:
Fixed.